### PR TITLE
[Doc] Add configs for broker

### PR DESCRIPTION
**Pull request by Anonymitaet**
Fix https://github.com/apache/pulsar/issues/5223

Add two configurations for broker:
* `allowAutoTopicCreationType`
* `allowAutoTopicCreationNumPartitions` (I have confirmed with @fxbing, this property has been renamed to `defaultNumPartitions`)

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@fxbing could you please help review? Thank you

File: site2/docs/reference-configuration.md [hunk 1]
```diff
@@ -126,9 +127,9 @@ Pulsar brokers are responsible for handling incoming messages from producers, di
 |backlogQuotaCheckEnabled|  Enable backlog quota check. Enforces action on topic when the quota is reached  |true|
 |backlogQuotaCheckIntervalInSeconds|  How often to check for topics that have reached the quota |60|
 |backlogQuotaDefaultLimitGB|  Default per-topic backlog quota limit |10|
-|allowAutoTopicCreation| Enable topic auto creation if new producer or consumer connected |true|
-|allowAutoTopicCreationType| The type of topic that is allowed to be automatically created.(partitioned/non-partitioned) |non-partitioned| 
-|defaultNumPartitions| The number of partitioned topics that is allowed to be automatically created if allowAutoTopicCreationType is partitioned |1|
+|allowAutoTopicCreation| Enable topic auto creation if a new producer or consumer connected |true|
+|allowAutoTopicCreationType| The topic type (partitioned or non-partitioned) that is allowed to be automatically created. |non-partitioned|
```
Comment (fxbing) [inline site2/docs/reference-configuration.md]: I have forgotten to modify the content I wrote before. Now the default type should be partitioned.

**Comment by fxbing** [comment]
Overall looks good to me, just left a comment. PTAL @Anonymitaet

File: site2/docs/reference-configuration.md (hunk 1 above)
Comment (Anonymitaet) [reply site2/docs/reference-configuration.md]: Thank you, I've udpated

**Comment by Anonymitaet** [comment]
run integration tests

**Comment by Anonymitaet** [comment]
run integration tests

**Comment by Anonymitaet** [comment]
run integration tests

**Comment by Anonymitaet** [comment]
run integration tests

**Comment by Anonymitaet** [comment]
run integration tests

**Comment by Anonymitaet** [comment]
run integration tests

**Comment by Anonymitaet** [comment]
run integration tests

**Pull request merged**